lib/utilities/get-config.js

```diff
@@ -3,11 +3,22 @@
 let Yam = require('yam');
 let Project = require('../models/project');
 
-module.exports = function getConfig(configOverride) {
-  if (configOverride) {
-    return configOverride;
-  }
+function generateConfig() {
   return new Yam('ember-cli', {
     primary: Project.getProjectRoot(),
   });
+}
+
+let config = generateConfig();
+
+module.exports = function getConfig(configOverride, shouldRegenerateConfig) {
```

Comment:
instead of adding another parameter for this, what do you think about exporting a separate reset function for tests?

in es6 this would be a non default export, but in this case it should be fine to hang it off the default export.  eg


```js

let config;

module.exports = function getConfig(configOverride) {
  if (configOverride) {
    return configOverride;
  }

  if (config === undefined) {
    config = generateConfig();
  }

  return config;
}

// private; used in testing
module.exports._reset = function() {
  config = undefined;
}
```

Reply:
I'll give that a shot. Thanks 👍

Reply:
@hjdivad it turns out the problematic tests were actually removed in a recent commit 😄 So there's no longer a need to jump through hoops to get the suite to pass. Tests are 🍏 locally and the config memoization works.

Reply:
oh nice; so if you rebase and push we should expect the PR to be green then

Reply:
it looks like master is still 🔴

Reply:
I think these tests we just moved into ember-source so that is why we're seeing the failures over there now when upgrading ember-cli.

@patocallaghan if it suits, let's have a chat about this tomorrow